compiler/stage.cpp

```diff
@@ -49,14 +46,11 @@ void on_compilation_error(const char *description __attribute__((unused)), const
   if (assert_level == WRN_ASSERT_LEVEL && warning_file) {
     file = warning_file;
   }
-  fmt_fprintf(file, "\n~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n{} [gen by {}:{}]\n", get_assert_level_desc(assert_level), file_name, line_number);
-  stage::print(file);
-  string correct_description;
-  if (!stage::should_be_colored(file)) {
-    correct_description = TermStringFormat::remove_special_symbols(full_description);
-  } else {
-    correct_description = full_description;
-  }
+  std::string error_title = get_assert_level_desc(assert_level);
+  fmt_fprintf(file, "\n~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n{} at stage: {}, gen by {}:{}\n",
+              stage::should_be_colored(file) ? TermStringFormat::paint_red(error_title) : error_title, stage::get_name(), file_name, line_number);
```

Comment:
I suggest write only file name (without full path) for 'gen by'.
Please, consider to use `kbasename(file_name)` for that.